### PR TITLE
types: Improve Convert function

### PR DESCRIPTION
**Pull request by shenli**
Check unsignedflag, charsetbin.
Remove useless code in cast builtin function.

### CODE REVIEW

File: util/types/convert.go [hunk 1]
```diff
@@ -127,6 +136,10 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		if err != nil {
 			return InvConv(val, tp)
 		}
+		// TODO: We should first convert to uint64 then check unsigned flag.
+		if mysql.HasUnsignedFlag(target.Flag) {
```
Comment (siddontang) [inline util/types/convert.go]: how about negative number?

File: util/types/convert.go [hunk 2]
```diff
@@ -71,6 +72,10 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		}
 		// TODO: consider target.Charset/Collate
 		x = truncateStr(x, target.Flen)
+		if target.Charset == charset.CharsetBin {
+			bx := []byte(x)
+			return bx, nil
```
Comment (qiuyesuifeng) [inline util/types/convert.go]: use `return []byte(x), nil`?

File: util/types/convert.go (hunk 1 above)
Comment (shenli) [inline util/types/convert.go]: Negative number will be converted to a big uint.

mysql> select cast(-5 as unsigned);
+----------------------+
| cast(-5 as unsigned) |
+----------------------+
| 18446744073709551611 |
+----------------------+
1 row in set (0.00 sec)

mysql>

**Comment by siddontang** [comment]
LGTM

File: util/types/convert.go [hunk 3]
```diff
@@ -127,6 +136,10 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		if err != nil {
 			return InvConv(val, tp)
 		}
+		// TODO: We should first convert to uint64 then check unsigned flag.
```
Comment (qiuyesuifeng) [inline util/types/convert.go]: first -> firstly?

**Comment by shenli** [comment]
PTAL @qiuyesuifeng

**Pull request merged**